CMakeLists.txt

```diff
@@ -234,7 +234,12 @@ if(ENABLE_MANUAL)
   endif()
 endif()
 # Required for building manual, docs, tests
-find_package(Perl REQUIRED)
+find_package(Perl)
+if(NOT PERL_FOUND)
+  set(ENABLE_MANUAL OFF)
```

Comment:
I would not override `ENABLE_MANUAL` since that is the user option to request the feature. Disabling `USE_MANUAL` and then check for that would be preferable I think.

Would it be an idea to warn that certain features are unavailable even if they were requested?

Reply:
@Lekensteyn Well, maybe it is a good idea, but the whole point is to not to use ENABLE_MANUAL at all if it did not satisfy the requirements even if enabled.

The warning is not made sense to me here, because if Perl not found - standard module always warns you.

Reply:
IMHO the right way would be: if the user specifically requested the manual bu we cannot generate it (because of missing Perl in this case), it's a hard, "cmake-time" error.

Reply:
@jzakrzewski It is not easy to check whether the user forced an option to be on or whether it was a default value (well, maybe if you check for cached variables).

To set a default based on feature availability, you would need something like this:
```cmake
find_package(Perl)
if(PERL_FOUND AND ...)
    set(ENABLE_MANUAL_DEFAULT ON)
else()
    set(ENABLE_MANUAL_DEFAULT OFF)
endif()
option(ENABLE_MANUAL "description here" ${ENABLE_MANUAL_DEFAULT})
```

@snikulov `ENABLE_xxx` should control `USE_xxx` directly. All others that depend on the feature should check for `USE_xxx` and not `ENABLE_xxx`. E.g. I would use this:
```
if(USE_MANUAL)
    add_subdirectory(docs)
endif()
```

Reply:
@Lekensteyn I know it's not trivial but this is what I as a user expect. I'm curious if there is some wider-spred method to do it.

Nevertheless, I agree with your review points.

Reply:
@Lekensteyn @jzakrzewski what about our agreement to use CURL_ prefix for options?
Should I name it CURL_ENABLE_... and CURL_USE...?

Reply:
@snikulov The command line options is the place where I would really leave the CURL_ prefix out. It's hell lot of typing without any benefit. The CMake options are already prefixed with CMAKE_, so no conflict there. After all, the user knows he's building curl - he/she doesn't have to type it 10 times on the command line. Just my opinion on the subject but I'm not gonna say "don't do this" ;)